Bump: Release 5.0.0 (#4638)
aristanetworks/ansible-avd: ansible_collections/arista/avd/docs/release-notes/5.x.x.md

Reviewer: * Line 13 points to `devel` for 5.x.x release documentation. May need an adjustment

> - Documentation for AVD version `5.x.x` [available here](https://avd.arista.com/devel/)

* Line 21 may be missing something: `"cvprac>=1.4.0" --upgrade. <HERE> has been removed`

> The latest version can be installed with `pip install "cvprac>=1.4.0" --upgrade`. has been removed from the `arista.avd` Ansible collection in v5.0.0.

* Line 70
```diff
-If additional custom keys are desired, a key starting with an underscore `_`, will be ignored.
+If additional custom keys are desired, a key starting with an underscore `_` will be ignored.
```

* Line 144. WOuld make sense to explicitly state what variable that was?
```diff
-In the previous implementation, loose mode was used by default to avoid stopping the playbook and hence the tests when they were run task by task in Ansible. It was possible to turn loose mode off with a variable that would make the playbook stop at the first failing tasks for whatever reason on each host.
+In the previous implementation, loose mode was used by default to avoid stopping the playbook and hence the tests when they were run task by task in Ansible. It was possible to turn loose mode off with a variable `eos_validate_state_validation_mode_loose: false` that would make the playbook stop at the first failing tasks for whatever reason on each host.
```

* LIne 176
> Starting from version 4.30.1F, `service_routing_protocols_model` is set to `multi-agent` by default on EOS devices.

Wouldn't be better to not reference AVD variables in regards to default EoS configuration?

```diff
-Starting from version 4.30.1F, `service_routing_protocols_model` is set to `multi-agent` by default on EOS devices.
+Starting from version 4.30.1F, `default service routing protocols` is set to `multi-agent` by default on EOS devices. Details are available [here](https://www.arista.com/en/support/toi/eos-4-30-1f/17697-use-multi-agent-as-default-routing-protocols-model) (TOI article is only available for registered users).
```

Author: - Line 13: has been updated to point to stable, which will be 5.0.0 when we release. I updated `release-notes/4.x.x.md` to point to https://avd.arista.com/4.10/
- Line 21: this was accidentally introduced in a PR due to rebasing. This was removed.
- Line 70: fixed
- Line 144: agreed and updated
- Line 176: agreed and updated

see commit: b114c44

See the [porting guide](../porting-guides/5.x.x.md#data-model-changes-from-dict-of-dicts-to-list-of-dicts) for details.

### All eos_designs and eos_cli_config_gen variables are always validated by schemas

In AVD 5.0.0, all `eos_designs` and `eos_cli_config_gen` keys are validated by schemas and execution will stop for any violations.

> If additional custom keys are desired, a key starting with an underscore `_` will be ignored.

Reviewer 2: ```suggestion
If additional custom keys are desired, a key starting with an underscore `_` will be ignored by validation.
```

| [The base class for custom interface descriptions `AvdInterfaceDescriptions` was moved](../porting-guides/5.x.x.md#the-base-class-for-custom-interface-descriptions-avdinterfacedescriptions-was-moved) |
| [Base class for custom IP addressing `AvdIpAddressing` was moved](../porting-guides/5.x.x.md#base-class-for-custom-ip-addressing-avdipaddressing-was-moved) |

### Breaking or behavioral changes in eos_cli_config_gen

Breaking changes may require modifications to the inventory or playbook. See the [Porting guide for AVD 5.x.x](../porting-guides/5.x.x.md#changes-to-role-aristaavdeos_cli_config_gen)
for details.

| Detailed changes and link to porting guide |
| ------------------------------------------ |
| [Removal of deprecated eos\_cli\_config\_gen data models](../porting-guides/5.x.x.md#removal-of-deprecated-eos_cli_config_gen-data-models) |
| [`!RANCID-CONTENT-TYPE: arista` is no longer at the top of the generated configurations](../porting-guides/5.x.x.md#rancid-content-type-arista-is-no-longer-at-the-top-of-the-generated-configurations) |

> | [No auto-conversion of old data model for `router_bgp.redistribute_routes` and `router_bgp.vrfs[].redistribute_routes`](../porting-guides/5.x.x.md#no-auto-conversion-of-old-data-model-for-router_bgpredistribute_routes-and-router_bgpvrfsredistribute_routes) |

Reviewer 2: This name of breaking change looks weird - is it a breaking change that we do not auto convert?

This is the global change of not auto converting dict of dicts to list of dicts anymore so not sure if this should even be in porting guide for this key in particular

> | [Default EOS configuration is no longer automatically generated](../porting-guides/5.x.x.md#default-eos-configuration-is-no-longer-automatically-generated) |

Reviewer 2: would group this one with RANCID

| [`ip_security.sa_policies[].esp.integrity` and `.encryption` `null` option has been replaced with `disabled`](../porting-guides/5.x.x.md#ip_securitysa_policiesespintegrity-and-encryption-null-option-has-been-replaced-with-disabled) |
| [`router_ospf.redistribute.bgp/connected/static` data model has been updated to require the `enabled` key](../porting-guides/5.x.x.md#router_ospfredistributebgpconnectedstatic-data-model-has-been-updated-to-require-the-enabled-key) |
| [`router_traffic_engineering.enabled` is required](../porting-guides/5.x.x.md#router_traffic_engineeringenabled-is-required) |
| [Removal of `type` key dependency for rendering ethernet/port-channel interfaces configuration and documentation](../porting-guides/5.x.x.md#removal-of-type-key-dependency-for-rendering-ethernetport-channel-interfaces-configuration-and-documentation) |
| [Data models for BGP additional-paths under `router_bgp` has been changed](../porting-guides/5.x.x.md#data-models-for-bgp-additional-paths-under-router_bgp-has-been-changed) |

> | [Removal of deprecated data models](../porting-guides/5.x.x.md#removal-of-deprecated-data-models-1) |

Reviewer: Link `5.x.x.md#removal-of-deprecated-data-models-1` seems to be incorrect
```diff
-| [Removal of deprecated data models](../porting-guides/5.x.x.md#removal-of-deprecated-data-models-1) |
+| [Removal of deprecated data models](../porting-guides/5.x.x.md#removal-of-deprecated-data-models_1) |
```

Author: fixed renamed the link so we don't have duplicates

#### Undeployed devices

> - All tests will be removed from the catalog for a device flagged as undeployed using the host level variable [`is_deployed: false`](https://avd.arista.com/en/stable/roles/eos_designs/docs/input-variables.html#flagging-a-device-as-not-deployed). Additionally, all tests take into account the `is_deployed` variable value and remove tests accordingly.

Reviewer: Aren't both sentences in this block saying the same thing?

Reviewer 3: It is missing a nuance in the second sentence around this being taken into account for peer-devices. So we don't test link state and ping etc towards a peer that is not deployed.

Reviewer 2: ```suggestion
- All tests will be removed from the catalog for a device flagged as undeployed using the host level variable [`is_deployed: false`](https://avd.arista.com/en/stable/roles/eos_designs/docs/input-variables.html#flagging-a-device-as-not-deployed).
- Additionally, the per-device generated test catalog takes into account the `is_deployed` variable value from the peer devices (e.g. a ping test towards an undeployed peer is not generated).
```

- BGP tests will only run if `service_routing_protocols_model` is set to `multi-agent` in the structured configuration file.

!!! note

>     Starting from version 4.30.1F, `default service routing protocols` is set to `multi-agent` by default on EOS devices. Details are available [here](https://www.arista.com/en/support/toi/eos-4-30-1f/17697-use-multi-agent-as-default-routing-protocols-model) (TOI article is only available for registered users).

Reviewer 2: ```suggestion
    Starting from version 4.30.1F, the service routing protocols is set to `multi-agent` by default on EOS devices by default. Details are available [here](https://www.arista.com/en/support/toi/eos-4-30-1f/17697-use-multi-agent-as-default-routing-protocols-model) (TOI article is only available for registered users).
```

Reviewer 4: Starting from version 4.30.1F, the service routing protocols is set to `multi-agent` by default on EOS devices` by default.`

Reviewer 4: `by default` is repeated in suggestion.

Author: resolved
